### ISSUE TITLE
NuGet.VisualStudio for VS2015?

### ISSUE THREAD

**Issue by tonysneed** [NuGet/Home#598]
I have a VSIX for VS2013 that installs project templates.  These templates use a wizard for collecting some user input, and the wizard also uses the [NuGet.VisualStudio NuGet package](http://www.nuget.org/packages/NuGet.VisualStudio/) for installing NuGet packages from the VSIX, as described in this article: https://docs.nuget.org/create/packages-in-visual-studio-templates.

I've updating my VSIX for compatibility with VS2015, but my wizard no longer installs any NuGet packages.  Is there an update to NuGet.VisualStudio which I could use to install NuGet packages with my project templates in VS2015?

**Comment by yishaigalatzer** [comment]
Hey @tonysneed nothing having changed, the API is still the same, and is being actively used by other wizards.

Can you give us some more details on what version of Visual Studio 2015 are you using? And can you show a code sample/simple repro?

**Comment by tonysneed** [comment]
VsTemplateWizard seemed to work for me with VS2015 CTP6, but it's not installing the packages for me with VS2015 RC.  I'll create a small repro. Thanks!

**Comment by tonysneed** [comment]
Alright, I created the repro and published it to GitHub:
https://github.com/tonysneed/VsTemplateWizardRepro

I tried creating a VSIX that includes a project template for installing Json.Net, *without using a custom wizard*.  This is based on the following article: [Authoring Visual Studio Templates with NuGet Dependencies](https://docs.nuget.org/create/packages-in-visual-studio-templates)

The VSIX targets both VS2013 or greater, so it also installs on VS2015RC.  After installing the VSIX, creating a project based on the template for VS2013 does install Json.Net, but creating one for VS2015RC does NOT install Json.Net.  I am using Community edition for both versions of VS.

**Comment by yishaigalatzer** [comment]
We will take a look. But this seems off. Can you try targeting dev14 only? We can take a deep look on Monday

**Comment by yishaigalatzer** [comment]
By the way, since you did such an awesome job on mvc PRs, we will gladly take one here as well :)

**Comment by tonysneed** [comment]
Yes, I tried again after [re-targeting](https://github.com/tonysneed/VsTemplateWizardRepro/tree/target-vs14) to VS 14 only, and I still get the same behavior: no packages are installed creating a new project with the template installed by the VSIX.

@yishaigalatze, I thought I recognized you!  Happy to help here as well. :smile:

**Comment by tonysneed** [comment]
Let me know if you can reproduce my repro.  Just to be sure, I think I'll create a new VM with a clean install of VS 2015 RC and see if I can repro it there.

**Comment by yishaigalatzer** [comment]
Sure, trying to go through this weekend without firing up my work laptop. Lets see how long I can last :) 

A clean VM sounds like a good idea.

**Comment by tonysneed** [comment]
Update - I created a virgin win 8.1 VM and installed VS 2015 RC and the SDK.  And I'm seeing the same behavior.  I have a VSIX that includes the Json.Net Nuget package and a project template with the following addition:

```xml
  <WizardExtension>
    <Assembly>NuGet.VisualStudio.Interop, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a</Assembly>
    <FullClassName>NuGet.VisualStudio.TemplateWizard</FullClassName>
  </WizardExtension>
  <WizardData>
    <packages repository="extension"
              repositoryId="SomeAppTemplate.Deploy.Anthony Sneed.65d863a8-762c-46c1-9f85-ee7d8d436e51">
      <package id="Newtonsoft.Json" version="6.0.8" />
    </packages>
  </WizardData>
```

When the VSIX is installed on VS2015 RC, new projects created from this template do not have the Json.Net package installed.  I'm not seeing this happen with either VS2013 or VS2015 CTP6.  So obviously something happened with RC. :confounded:

**Comment by yishaigalatzer** [comment]
Hey Tony - Can you please share the actual VSIX somewhere? The sample itself is not sufficient to get a repro (at least not easily).

This seems like an important issue so I'd like to get it into RTM as soon as possible.

**Comment by yishaigalatzer** [comment]
@tonysneed the template you are providing is not following the guidelines in the document you share above. You are including the packages.config and the references in the template itself. Where the guidance says to not do it.

Once I removed them things start working for me. So this seems like perhaps a behavior change, but it is still within the spec.

I prefer not to try and fix it, can you please verify my findings on your side?

**Comment by tonysneed** [comment]
Let me give it a try. It used to be you would have to Indicate the location of packages in the vstemplate file, in the WizardData section. Didn't realize that had changed.

**Comment by tonysneed** [comment]
I see now ... here is the relevant paragraph:

> You should author your template such that it does not include the packages.config file or any references or content that installing the NuGet package(s) would add to the project. In the next section, you'll use a custom wizard that will add those files and references to the project as part of template expansion.

I'll update the project template to exclude packages.config and references, and I'll let you know if that helps.

**Comment by yishaigalatzer** [comment]
I made the change on my side (remove packages.config and references from the project) and it worked for me. I had to build my own vsix.

**Comment by tonysneed** [comment]
Alright, this works for me now. It looks like the omission of packages.config and references was not enforced prior to VS2015 RC.  It's a tad inconvenient because it requires modifying the csproj and vstemplate files to remove those items, then re-zip the template.  But as you mentioned, that's what the docs say to do. :smiley: 

Yes, the option I selected was to *embed the NuGet packages in the VSIX*, which is generated by building the .Deploy project in my solution.  The VSIX is placed in the **bin\debug** directory, and running it installs the  VSIX with the project template.  For good measure, I placed the generated VSIX in the GitHub repo.  I'll push my changes to the repo and test with VS2013 as well.
